syntax = "proto3";

message OrderBook
{
   enum OrderWay
   {
      buy = 0;
      sell = 1;
   }

    message Statistics
    {
        double last_price = 1;
        double high_price = 2;
        double low_price  = 3;
    }

   message Order
   {
      uint64   identifier        = 1;
      OrderWay way               = 2;
      double   quantity          = 3;
      double   canceled_quantity = 4;
      double   executed_quantity = 5;
      double   price             = 6;
      string   counterparty      = 7;
      uint64   timestamp         = 8;
   }

   uint64 instrument_identifier = 1;
   Statistics statistics        = 2;
   repeated Order orders        = 3;
}